pkg/agent/util/net.go

```diff
@@ -100,6 +101,32 @@ func dialUnix(address string) (net.Conn, error) {
 	return net.Dial("unix", address)
 }
 
+func GetIPNetDeviceByName(ifaceName string, ovsBridgeName string) (*net.IPNet, *net.Interface, error) {
+	link, err := net.InterfaceByName(ifaceName)
+	if err != nil {
+		if !strings.Contains(err.Error(), "no such network interface") {
+			return nil, nil, err
+		}
+		link, err = GetVirtualInterfaceByName(ifaceName, ovsBridgeName)
```

Comment:
This process is not generic enough to be a common util.

Reply:
Do you mean implement this function in both net_windows and net_others?

Reply:
As an util function, I think `GetIPNetDeviceByName` is too specific to its client's usage that it would first search interface with the name, then search interface with "vEthernet (%s)", then search interface with ovsBridgeName.
It makes the function hard to reuse in the future while it occupies the generic name "GetIPNetDeviceByName" if a real "GetIPNetDeviceByName" is needed.

Reply:
Got your point, I would rename the function as "GetTransportIPNetDeviceByName", what do you think ?

Reply:
replied in https://github.com/antrea-io/antrea/pull/2370#discussion_r674949031